src/sync/sync_user.cpp

```diff
@@ -175,11 +175,11 @@ void SyncUser::update_refresh_token(std::string token)
             case State::Removed:
                 return;
             case State::LoggedIn:
-                m_refresh_token = token;
+                m_refresh_token = std::move(token);
                 break;
             case State::LoggedOut: {
                 sessions_to_revive.reserve(m_waiting_sessions.size());
-                m_refresh_token = token;
+                m_refresh_token = std::move(token);
```

Comment:
I would also change `update_refresh_token(std::string token)` to be `update_refresh_token(std::string& token)` since the only place this is called already receives a copy of the token.

Reply:
`update_refresh_token(std::string token)` has been changed to `update_refresh_token(std::string&& token)` as `SyncManager::get_user` starts the move semantic meta-game and is the first source to call `update_refresh_token `